homeassistant/components/mqtt/light/schema_json.py

```diff
@@ -175,6 +175,43 @@ def _setup_from_config(self, config):
         self._supported_features |= config[CONF_XY] and SUPPORT_COLOR
         self._supported_features |= config[CONF_HS] and SUPPORT_COLOR
 
+    def _parse_color(self, values):
+        try:
+            red = int(values["color"]["r"])
+            green = int(values["color"]["g"])
+            blue = int(values["color"]["b"])
+
+            return color_util.color_RGB_to_hs(red, green, blue)
+        except KeyError:
+            pass
+        except ValueError:
+            _LOGGER.warning("Invalid RGB color value received")
+            return self._hs
+
+        try:
+            x_color = float(values["color"]["x"])
+            y_color = float(values["color"]["y"])
+
+            return color_util.color_xy_to_hs(x_color, y_color)
+        except KeyError:
+            pass
+        except ValueError:
+            _LOGGER.warning("Invalid XY color value received")
+            return self._hs
```

Comment:
Should we raise to avoid further execution? We got invalid data.

Reply:
I don't know, that would be a breaking change if anyone happens to depend on it not raising.
Could we get this into 2021.2, then maybe be a bit more strict in a separate PR to give some time to react to that?